### PR TITLE
Deprecation Policy

### PR DESCRIPTION
**Pull request by jasonblais**
@lindy65 You don't need to queue for PM review, I'll review the doc during PM standup.

### CODE REVIEW

**Comment by jasonblais** [comment]
@jwilander Let me know what you think of this proposal

File: source/process/deprecated-features.rst [hunk 1]
```diff
@@ -1,14 +1,60 @@
-Announcing Deprecated Features
+Deprecation Policy
 ===============================
 
-This document outlines the process for announcing deprecated features to the community. The guiding principle is `no surprises <https://docs.mattermost.com/developer/manifesto.html#no-surprises>`_, where users should never run into anything unexpected with Mattermost.
+This document outlines the process for announcing deprecated features to the community. The guiding principle is `no surprises <https://docs.mattermost.com/developer/manifesto.html#no-surprises>`_ with guaranteed long-term stability, where admins or users should never run into anything unexpected with Mattermost.
```
Comment (lindy65) [inline source/process/deprecated-features.rst]: "...with Mattermost." > "...while using Mattermost." ?

File: source/process/deprecated-features.rst [hunk 2]
```diff
@@ -1,14 +1,60 @@
...
+
++-----------------------+---------------------+---------------------+
+| Deprecation Announced | Final Minor Release | Removal Target Date |
++=======================+=====================+=====================+
+| 3.9.0                 | 3.10.0              | 4.0.0               |
++-----------------------+---------------------+---------------------+
+| 3.10.0                | 3.10.0              | 5.0.0               |
++-----------------------+---------------------+---------------------+
+
+Exceptions for the removal target date may be made if it impacts security or performance of using Mattermost. In such cases, the target date for removing the feature may be made sooner.
```
Comment (lindy65) [inline source/process/deprecated-features.rst]: "...performance of using Mattermost." > "...the performance of Mattermost." ?

**Comment by jasonblais** [comment]
PM approved

**Comment by lindy65** [comment]
@jwilander, should this be reviewed in a dev meeting so all devs are aware?  JB reviewed with PMs as a group.  Another option would be Platform meeting?

**Comment by lindy65** [comment]
JW will discuss in dev meeting.

File: source/process/deprecated-features.rst [hunk 3]
```diff
@@ -1,14 +1,60 @@
...
+2. `12 working days before each release <https://docs.mattermost.com/process/release-process.html#c-t-minus-12-working-days-cut-off-for-merging-major-features>`_, sends the list of deprecated features to the marketing manager, who includes this information in the release announcement.
+3. `2 working days before each release <https://docs.mattermost.com/process/release-process.html#h-t-minus-2-working-days-release-build-cut>`_, ensures the `deprecated features page <https://about.mattermost.com/deprecated-features/>`_ is up to date.
+
+Removal target date
+~~~~~~~~~~~~~~~~~~~~~
+
+The removal target date should always be the date of the next major release, such as v4.0.0. If the date is not known, you can reference the next major version rather than the actual release date.
+
+However, there should always be at least two months from the time the deprecation is announced to its removal. This number is chosen to match our security backport release policy.
```
Comment (jwilander) [inline source/process/deprecated-features.rst]: Wondering if two months is a bit too short? GitLab does 3 months right?

Comment (jasonblais) [reply source/process/deprecated-features.rst]: GitLab has 2+ months, except 3 months for sensitive changes that can cause major service outage

**Pull request merged**